### ISSUE TITLE
linter: false-positive in `eslint-plugin-unicorn(no-invalid-fetch-options)`

### ISSUE THREAD

**Issue by GameRoMan** [oxc-project/oxc#16569]
### What version of Oxlint are you using?

Not sure, I am using VS Code etension ` oxc.oxc-vscode` 1.31.0

### What command did you run?

I am using VS Code extension

### What does your `.oxlintrc.json` config file look like?

No config, just a VS Code extension

### What happened?

Minimal reproduction example

```ts
declare const logic: boolean;
```
```js
fetch("/url", {
  method: logic ? "PATCH" : "POST",
  body: "some body",
});
```

> `"body" is not allowed when method is "GET"`

<img width="761" height="29" alt="Image" src="https://github.com/user-attachments/assets/0b43c071-d25a-45cc-be50-e7b2167bc0b3" />

**Comment by gameroman** [comment]
Wow, thank you, that was very fast
